apps/site/pages/en/learn/modules/how-to-use-streams.md

```diff
@@ -0,0 +1,599 @@
+
+### Streams 0
+
+The first version of streams was released at the same time as Node.js. Although there wasn't a Stream class yet, different modules used the concept and implemented the `read`/`write` functions. `util.pump()` function was available to control the flow of data between streams.
+
+### Streams 1 (Classic)
+
+With the release of Node v0.4.0 in 2011 the Stream class was introduced as well as the `pipe()` method.
+
+### Streams 2
+
+In 2012, with the release of Node v0.10.0, Streams 2 were unveiled. This update brought new stream subclasses, including Readable, Writable, Duplex, and Transform. Additionally, the `readable` event was added. To maintain backwards compatibility, streams could be switched to the old mode by adding a `data` event listener or calling `pause()` or `resume()` methods.
+
+### Streams 3
+
+In 2013, Streams 3 were released with Node v0.11.5, to address the problem of a stream having both a `data` and `readable` event handlers. This removed the need to choose between 'current' and 'old' modes. Streams3 is the current version of streams in Node.js.
+
+## Stream types
+
+### Readable
+
+[`Readable`][] is the class that we use to sequentially read a source of data. Typical examples of `Readable` streams in Node.js API are [`fs.ReadStream` ][] when reading files, [`http.IncomingMessage` ][] when reading HTTP requests, and [`process.stdin` ][] when reading from the standard input
+
+#### Key Methods and Events
+
+A readable stream operates with several core methods and events that allow fine control over data handling:
+
+- **[`on('data')`][]**: This event is triggered whenever data is available from the stream. It is very fast, as the stream pushes data as quickly as it can handle, making it suitable for high-throughput scenarios.
+- **[`on('end')`][]**: Emitted when there is no more data to read from the stream. It signifies the completion of data delivery. This event is only fired when all the data from the stream has been consumed.
+- **[`on('readable')`][]**: This event is triggered when there is data available to read from the stream or when the end of the stream has been reached. It allows for more controlled data reading when needed.
+- **[`on('close')`][]**: This event is emitted when the stream and its underlying resources have been closed and indicates no more events will be emitted.
+- **[`on('error')`][]**: This event can be emitted at any point signalling that there was an error processing. A handler of this event can be used to avoid uncaught exceptions
+
+A demonstration of the use of this events can be seen in the following sections
+
+#### Basic Readable Stream
+
+Here's an example of a simple readable stream implementation that generates data dynamically:
+
+```cjs
```

Comment:
Some examples are in CommonJS, while others are in ECMASCript Module Syntax, can you choose one or the other?

Reply:
Note that you can use both. Two codeboxes one after the other can be put together to make a code tab.

Reply:
Oh, yes, I intended to have both everywhere, but I was waiting until they were reviewed so I didn't have to change things twice. I'll update them tomorrow.

Reply:
Done!